mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreRightManagerTest.java

```diff
@@ -24,17 +24,176 @@
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+    private MailboxMapper mockedMailboxMapper;
+
+    @Before
+    public void setup() throws MailboxException {
+        alice = "Alice";
+        aliceSession = new MockMailboxSession(alice);
+        MailboxSessionMapperFactory mockedMapperFactory = mock(MailboxSessionMapperFactory.class);
+        mockedMailboxMapper = mock(MailboxMapper.class);
+        mailboxAclResolver = new UnionMailboxACLResolver();
+        groupMembershipResolver = new SimpleGroupMembershipResolver();
+        when(mockedMapperFactory.getMailboxMapper(aliceSession))
+            .thenReturn(mockedMailboxMapper);
+        storeRightManager = new StoreRightManager(mockedMapperFactory,
+                                                  mailboxAclResolver,
+                                                  groupMembershipResolver);
+    }
+
+    @Test
+    public void hasRighShouldThrowMailboxNotFoundExceptionWhenMailboxDoesNotExist() throws MailboxException {
+        expectedException.expect(MailboxNotFoundException.class);
+
+        MailboxPath mailboxPath = MailboxPath.forUser(alice, "unexisting mailbox");
+        when(mockedMailboxMapper.findMailboxByPath(mailboxPath)).thenThrow(MailboxNotFoundException.class);
+        storeRightManager.hasRight(mailboxPath, Right.Read, aliceSession);
+    }
+
+    @Test
+    public void hasRighShouldReturnTrueWhenTheUserOwnTheMailbox() throws MailboxException {
+        Mailbox mailbox = mock(Mailbox.class);
+        when(mailbox.getUser()).thenReturn(alice);
+
+        assertThat(storeRightManager.hasRight(mailbox, Right.Write, aliceSession))
+            .isFalse();
+    }
+
+    @Test
+    public void hasRighShouldReturnTrueWhenTheUserDoesnotOwnTheMailboxButHaveTheCorrectRightOnIt() throws MailboxException {
+        Mailbox mailbox = mock(Mailbox.class);
+        when(mailbox.getUser()).thenReturn("bob");
+        when(mailbox.getACL()).thenReturn(new MailboxACL(new MailboxACL.Entry(alice, Right.Write)));
+
+        assertThat(storeRightManager.hasRight(mailbox, Right.Write, aliceSession))
+            .isTrue();
+    }
+
+    @Test
+    public void hasRighShouldReturnTrueWhenTheUserDoesnotOwnTheMailboxButHasAtLeastTheCorrectRightOnIt() throws MailboxException {
+        Mailbox mailbox = mock(Mailbox.class);
+        when(mailbox.getUser()).thenReturn("bob");
+        when(mailbox.getACL()).thenReturn(new MailboxACL(new MailboxACL.Entry(alice, Right.Write, Right.Lookup)));
+
+        assertThat(storeRightManager.hasRight(mailbox, Right.Write, aliceSession))
+            .isTrue();
+    }
+
+    @Test
+    public void hasRighShouldReturnFalseWhenTheUserDoesNotOwnTheMailboxAndHasNoRightOnIt() throws MailboxException {
+        Mailbox mailbox = mock(Mailbox.class);
+        when(mailbox.getUser()).thenReturn("bob");
+
+        assertThat(storeRightManager.hasRight(mailbox, Right.Write, aliceSession))
+            .isFalse();
+    }
```

Comment:
you can add a test hasRighShouldReturnFalseWhenTheUserDoesNotOwnTheMailboxAndHasIncorrectRightOnIt